### PR TITLE
Add link icon to anchor link headings

### LINKED ISSUE #209: Make subtitles linkable
I am think it would be helpful to have subtitles linkable from a main section.
https://10up.github.io/Engineering-Best-Practices/php/

If it make sense i will go ahead and create a PR for this

### PR DESCRIPTION
**Pull request by colorful-tones**
Fixes #209 

I used the existing IcoMoon `selection.json` and added the link icon to the set. I then used a pseudo-element to put an icon next to each `.anchor-heading` (existing class).

Demonstration screenshot of ending result:
screenshot 2019-01-15 09 43 53

### CODE REVIEW

File: _includes/_sass/includes/_docs-section.scss [hunk 1]
```diff
@@ -18,11 +18,24 @@
 	.anchor-heading {
 		padding-top: 60px;
 		margin-top: -60px;
+
+		&::after {
```
Comment (timwright12) [inline _includes/_sass/includes/_docs-section.scss]: Sorry, can you make this a clickable element in the DOM? We want to be able to click the icon so the user can get the sharable URL

**Review by timwright12 — CHANGES_REQUESTED** [review]

**Comment by colorful-tones** [comment]
I added a new Liquid Template partial, which accepts an `anchor="foo"` argument. This is used to output the anchor link in each template.

I then added the call with appropriate anchor link to each heading section, on each page. I then tested each page to verify that anchor links were matching.

Attaching Gif of resulting treatment:
10upengineering-anchor-links

**Review by timwright12 — CHANGES_REQUESTED** [review]
A few thing here:
**1. Let's bring back the link icon by default so people know you can link to the heading without hovering**

screen shot 2019-01-18 at 9 25 50 am

**2. On the actual hash, let's make sure it aligns with the "back to top" link (we can just align all 3 items to the baseline of the heading text) and sync those font sizes (make the hash smaller). Also we'll want the link icon to be center aligned with the hash text.**

screen shot 2019-01-18 at 9 23 05 am

**3. Check all your screen sizes to make sure we don't get weird breaks between the hash text and the back to top (I know this is dependent on the hash length, but let's do our best)**
 
screen shot 2019-01-18 at 9 22 50 am
screen shot 2019-01-18 at 9 22 36 am

**4. Make sure the hash can be exposed on focus as well as hover for keyboard users.**

**Review by timwright12 — CHANGES_REQUESTED** [review]
@colorful-tones 
**Visual**:
Lets enlarge the link icon to 1em x 1em, then center it vertically and remove the opacity so the color contrast doesn't dip below our accessibility standards.

**Interaction**:
The default state of the heading should be: [heading] [link icon] [back to top]
On hover and/or focus of the [link icon], display the [#hash] as well to make it: [heading] [#hash] [link icon] [back to top]

**Comment by colorful-tones** [comment]
> **Visual**:
Lets enlarge the link icon to 1em x 1em, then center it vertically and remove the opacity so the color contrast doesn't dip below our accessibility standards.

These changes have been implemented.

> **Interaction**:
The default state of the heading should be: [heading] [link icon] [back to top]
On hover and/or focus of the [link icon], display the [#hash] as well to make it: [heading] [#hash] [link icon] [back to top]

These changes have been implemented.

Gif of current appearance and outcome:
anchor-link

**Comment by colorful-tones** [comment]
Tagging @timwright12 for review please.

**Review by timwright12 — APPROVED** [review]

**Pull request merged**